indico/modules/rb_new/util.py

```diff
@@ -541,3 +549,36 @@ def serialize_nonbookable_periods(data):
 
 def serialize_unbookable_hours(data):
     return [bookable_hours_schema.dump(d).data for d in data]
+
+
+def build_rooms_spritesheet():
+    def _resize(image):
+        return image.resize(ROOM_PHOTO_DIMENSIONS, Image.ANTIALIAS)
+
+    image_width, image_height = ROOM_PHOTO_DIMENSIONS
+    query = Room.query.filter(Room.photo)
+    room_count = query.count()
+    mapping = []
+    sprite_width = (image_width * (room_count + 1))  # +1 for the placeholder
+    sprite_height = image_height
+    sprite = Image.new(
+        mode='RGB',
+        size=(sprite_width, sprite_height),
+        color=(0, 0, 0))
+
+    # Placeholder image at position 0
+    assets_dir = os.path.join(current_app.root_path, 'web', 'static')
+    no_photo_path = 'rooms/large_photos/NoPhoto.jpg'
+    no_photo_image = Image.open(os.path.join(assets_dir, config.IMAGES_BASE_URL[1:], no_photo_path))
```

Comment:
`IMAGES_BASE_URL` is not a good idea here; it may contain  e.g. `/indico/` if someone uses a subdir. just use `os.path.join(current_app.root_path, 'web/static/images/rooms/....')` directly